pkg/tempo/automaticloggingprocessor/factory.go

```diff
@@ -0,0 +1,70 @@
+package automaticloggingprocessor
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+// TypeStr is the unique identifier for the Automatic Logging processor.
+const TypeStr = "automatic_logging_processor"
```

Comment:
nit: `automatic_logging` would be more in line with the resto of the processors which don't include the processor suffix.